src/Paraunit/File/TempDirectory.php

```diff
@@ -42,7 +43,11 @@ public function getTempDirForThisExecution()
      */
     public static function getTempBaseDir()
     {
-        foreach (self::$tempDirs as $directory) {
+        $dirs = self::$tempDirs;
+        // Fallback to sys temp dir
+        $dirs[] = sys_get_temp_dir();
+
+        foreach ($dirs as $directory) {
             if (file_exists($directory)) {
                 $baseDir = $directory . DIRECTORY_SEPARATOR . 'paraunit';
                 self::mkdirIfNotExists($baseDir);
```

Comment:
Shouldn't we catch the exception here? We would like to try all the possible temp dirs before giving up, right?